Change Pairing.ComplementTable to pre-compute the table.

yasushi-saito opened the pull request:
It improves the performance of RevComp, etc.  It also reduces heap pressures.

Before:
BenchmarkRevComp-56      1000000              1518 ns/op                                                                                                      

After:
BenchmarkRevComp-56      2000000               704 ns/op                        

It causes the ComplementTable to return a shared table instance. All the uses found in biogo aren't affected by this change.

Commit:
Change Pairing.ComplementTable to pre-compute the table.

It improves the performance of RevComp, etc.  It also reduces heap pressures.

Commit:
Reformatted.

kortschak reviewed (changes requested):
Thanks. Would you please address the final commen and add yourself to A+C, and then I'll approve and merge.

Commit:
Added Saito to the A+C.

Clarify that the table returned by ComplementTable() shall not be mutated by the
caller.
